### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by Foxandxss** [snapcrunch/electron-preferences#76]
This tool is very very nice, thanks for building it.

Although, it has serious accessibility issues.

The placeholders are quite invisible:

![](https://i.imgur.com/J3PmfIu.png)

This foreground (#BDC2C9) with that background (#E7E7E7) gives a contrast ratio of 1.44:1 (it should be at least 4.5:1) See it [here](https://webaim.org/resources/contrastchecker/?fcolor=BDC2C9&bcolor=E7E7E7).

Happens the same in Dark mode

![](https://i.imgur.com/FNX2CW8.png)

It has slightly a better contrast ratio (2.32:1) as seen [here](https://webaim.org/resources/contrastchecker/?fcolor=5D5D5D&bcolor=252525)

----

The inputs (text, dropdown, list, perhaps more)  have no label so if a blind user tabs into it, it won't say what its supposed to be put in there.

One solution (easier one) is to change the `div` you have as a label for a proper label and then use `htmlFor` and an `id`. Another one, in certain cases is to use `aria-describedby` and point it to the help message.

----

There is no way to move to the preferences sidebar with the keyboard alone.

----

Perhaps there are more, and I understand this is always problematic, but I believe your library is really nice but this issues will prevent a good majority of apps from using it. There are laws that enforces companies to do it and this will discard electron-preferences.

**Comment by pvrobays** [comment]
Hi @Foxandxss. Thanks for reporting this issue.

TBH. I am not that familiar with accessibility optimizations, but happy to learn something new :)
You're mentioning there are laws enforcing companies to take into account accessibility optimizations, do you have any reference on this? I would love to read more about this.

Concerning the issue of changing the tabs in the sidebar via the keyboard. Have you an implementation in mind? What's the default fix for this? Is creating a simple <a> tag that's selectable via the tab-key fine?

I'm not sure if you have enough free time and motivation to do so, but would you like to contribute to this library and try to fix this issue yourself? 
If you were to find other major accessibility issues, don't mind adding them here.

**Comment by Foxandxss** [comment]
There are 3 levels of accessibility, A, AA, AAA. Most countries (if not all of them) enforces you to apply at least AA for government applications. Other countries will enforce it as well when the company has more than 100 employees. In reality many applications fails to apply this, but as a library creators, we have the responsibility to pave the way.

Colors are an easy fix. Check the website I shared with my examples and try to pick a combination that works for you visually and pass the AA check. A darker .help color will surely fix it.

For the inputs and labels, you have already a few of them using a proper label with `htmlFor`. It is just follow the same pattern.

The sidebar is more challenging but we can visualize it as a tab (I haven't properly check your code yet). The most common pattern for a tablist is to use an `ul` or `ol` and every tab item is a `li`. You can apply the role of `tablist` and `tabitem` respectively. That makes the sidebar tabbable and you can navigate through it with your cursors. Then with ENTER or ESPACE you can activate that section and tabbing again should move you to the section. This is done with aria-controls where you specify a relationship between the actual menu item and a section. Your current div solution is not tabbable and doesn't trigger anything with ENTER or ESPACE. Can be fixed with proper events for those keys, tabIndex and some aria-roles for tablist and tabitem. That or switch to lists.

That being said, I am no one to make you do the changes but I do consider it is a nice "excercise" to learn more how others navigates through an application.

Colors and labels should be easy. Navigation is a bit more challenging since it has more moving pieces.

I will add a reminder for myself about navigation and make a PR when I find some time to spare

**Comment by pvrobays** [comment]
FYI @Foxandxss created PR #86 for this issue.
The contrast of the text, and the issue of changing the tabs with the keyboard alone should've already been fixed.
Still need to add labels to the different fields/elements though.

Could you already check the branch (git branch: `feature/accesibility`) out and test it yourself + give feedback?
You don't happen to know an end-user with some 'disabilities' (definitely no offence intended) who can test the end result?

Thanks!

**Comment by pvrobays** [comment]
Closing the issue as the implementation has already been released. Other requests could be done in a new issue.
